core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -224,7 +227,9 @@ private class ShuffleStatus(
     logDebug(s"Removing existing map output ${mapIndex} ${bmAddress}")
     if (mapStatuses(mapIndex) != null && mapStatuses(mapIndex).location == bmAddress) {
       _numAvailableMapOutputs -= 1
-      mapStatusesDeleted(mapIndex) = mapStatuses(mapIndex)
+      val currentMapStatus = mapStatuses(mapIndex)
+      mapIdToMapIndex.remove(currentMapStatus.mapId)
```

Comment:
Removing it here will mean we cant query for it in `mapStatusesDeleted`, where we are relying on mapId -> mapIndex being in mapIdToMapIndex even when mapIndex is in `mapStatusesDeleted`

We should move this cleanup to when `mapStatusesDeleted` is being cleaned up.

Same applies to the cases below as well.

Reply:
This is a good point. But IIUC, `mapStatusesDeleted` will only be cleanedup when there is recovery happen using K8s. So it's not guaranteed to be always cleaned up in the end. I removed the dependency of  `mapIdToMapIndex` for `mapStatusesDeleted` as it's not a common use case.